### PR TITLE
Paying fee in currency other than Celo should be feasible if you just have enough.

### LINKED ISSUE #1663: Paying fee in currency other than Celo should be feasible if you just have enough.
## Description

https://www.notion.so/clabsco/Remove-quirks-list-HardFork-Changes-7ee3a0b904ea464db6af45aba9f94a4a#fcea747a4c0646aeae47b3496c8d477e

Paying fee in currency other than Celo(cUSD, cEUR, ...) should be feasible if you just have enough.

## Current Behavior

Currently, if you pay tx fee with CELO, having just enough amount in your account will be sufficient.
That's not the case for other currencies, for which you need strictly greater amount in your account.

## Expected Behavior

Paying tx fee with other currencies should be same as CELO.

### PR DESCRIPTION
**Pull request by YoduYodu**
### Description

Modified tx fee checks in `state_transition` and `tx_pool`.

### Tested

Automated tests.

### Related issues

- Fixes #1663 

### Backwards compatibility

Requires HF because this changed how transaction validity is checked

### CODE REVIEW

File: core/tx_pool.go [hunk 1]
```diff
@@ -1574,10 +1576,13 @@ func ValidateTransactorBalanceCoversTx(tx *types.Transaction, from common.Addres
 			return err
 		}
 
-		// To match the logic in canPayFee() state_transition.go, we require the balance to be strictly greater than the fee,
-		// which means we reject the transaction if balance <= fee
+		// This is required to match the logic in canPayFee() state_transition.go
+		// - Prior to E hardfork: we require the balance to be greater than or equal to the fee,
```
Comment (trianglesphere) [inline core/tx_pool.go:1580]: comments are not different between pre and post fork

**Comment by YoduYodu** [comment]
Messed up the commit history, will fix it soon

File: core/tx_pool.go [hunk 2]
```diff
@@ -1574,10 +1576,10 @@ func ValidateTransactorBalanceCoversTx(tx *types.Transaction, from common.Addres
 			return err
 		}
 
-		// To match the logic in canPayFee() state_transition.go, we require the balance to be strictly greater than the fee,
-		// which means we reject the transaction if balance <= fee
+		// To match the logic in canPayFee() state_transition.go, we require the balance to be greater than or equal to the fee,
```
Comment (trianglesphere) [inline core/tx_pool.go:1579]: Could you add comment about the HF behavior here?

**Review by trianglesphere — COMMENTED** [review]
Needs to have a merge in from master, but looking almost done.

File: core/tx_pool.go (hunk 2 above)
Comment (YoduYodu) [reply core/tx_pool.go:1579]: Oops, I misunderstood your prior comment.

**Review by trianglesphere — APPROVED** [review]
lgtm

**Pull request merged**